internal/databases/postgres/query_runner.go

```diff
@@ -180,19 +192,23 @@ func (queryRunner *PgQueryRunner) buildGetPhysicalSlotInfo() string {
 func (queryRunner *PgQueryRunner) getVersion() (err error) {
 	queryRunner.Mu.Lock()
 	defer queryRunner.Mu.Unlock()
+	pgCtx, cancel := ContextWithPgTimeout(queryRunner.Ctx, queryRunner.PgTimeout)
+	defer cancel()
```

Comment:
it seems that we can shorten code a bit by removing `defer cancel()` lines

Reply:
Hi @ostinru, thanks for the suggestion!

I just wanted to double-check: did you mean that we should remove defer cancel() entirely?

My understanding is that even if the context isn’t canceled explicitly, calling cancel() is still important to free up internal resources.